src/decode_call.rs

```diff
@@ -0,0 +1,61 @@
+use crate::map_err;
+use crate::types::DecodedSolValue;
+use anyhow::Context;
+use hypersync_client::format::{Data, Hex};
+use std::sync::Arc;
+
+#[napi]
+#[derive(Clone)]
+pub struct CallDecoder {
+    inner: Arc<hypersync_client::CallDecoder>,
+    checksummed_addresses: bool,
+}
+
+#[napi]
+impl CallDecoder {
+    #[napi]
+    pub fn from_signatures(signatures: Vec<String>) -> napi::Result<CallDecoder> {
+        let inner = hypersync_client::CallDecoder::from_signatures(&signatures)
+            .context("build inner decoder")
+            .map_err(map_err)?;
+
+        Ok(Self {
+            inner: Arc::new(inner),
+            checksummed_addresses: false,
+        })
+    }
+
+    pub fn enable_checksummed_addresses(&mut self) {
+        self.checksummed_addresses = true;
+    }
+
+    pub fn disable_checksummed_addresses(&mut self) {
+        self.checksummed_addresses = false;
+    }
```

Comment:
I personally prefer this to be passed in on construction than to be set afterwards, what do you think?

Reply:
We have the same logic for Decoder everywhere too
I don't have a strong opinion about that but changing it in Decoder will be breaking change, and having a different interface in CallDecoder would be weird 
We could try to accommodate both options.

Reply:
Yeah I've seen it, but it gets passed by config for instantiating the client. So both patterns currently exist. I would advocate for moving it to setting on construction everywhere but I understand that's not the scope of this and it's not blocking to getting this merged.

Reply:
Actually even changing the current interface (let's say we put it in from_sigs will break compatibility)

Reply:
We can add another constructor function like from_config_and_sigs() which takes config struct as well an has only one parameter for now. This won't be a breaking change. Can do it in next PR